src/plugins/fs/reiserfs.c

```diff
@@ -0,0 +1,386 @@
+ *
+ * Returns: (transfer full): information about the file system on @device or
+ *                           %NULL in case of error
+ *
+ * Tech category: %BD_FS_TECH_REISERFS-%BD_FS_TECH_MODE_QUERY
+ */
+BDFSReiserFSInfo* bd_fs_reiserfs_get_info (const gchar *device, GError **error) {
+    const gchar *args[3] = {"debugreiserfs", device, NULL};
+    gboolean success = FALSE;
+    gchar *output = NULL;
+    BDFSReiserFSInfo *ret = NULL;
+    gchar **lines = NULL;
+    gchar **line_p = NULL;
+    gchar *val_start = NULL;
+
+    if (!check_deps (&avail_deps, DEPS_DEBUGREISERFS_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    success = bd_utils_exec_and_capture_output (args, NULL, &output, error);
+    if (!success)
+        /* error is already populated */
+        return FALSE;
+
+    ret = g_new0 (BDFSReiserFSInfo, 1);
+
+    success = get_uuid_label (device, &(ret->uuid), &(ret->label), error);
+    if (!success) {
+        /* error is already populated */
+        bd_fs_reiserfs_info_free (ret);
+        return NULL;
+    }
+
+    lines = g_strsplit (output, "\n", 0);
+    g_free (output);
+    line_p = lines;
+    /* find the beginning of the (data) section we are interested in */
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Count of blocks on the device:"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Count of blocks on the device: 127744" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->block_count = g_ascii_strtoull (val_start, NULL, 0);
+
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Blocksize:"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Blocksize: 4096" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->block_size = g_ascii_strtoull (val_start, NULL, 0);
+
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Free blocks"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Free blocks (count of blocks - used [journal, bitmaps, data, reserved] blocks): 119529" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->free_blocks = g_ascii_strtoull (val_start, NULL, 0);
```

Comment:
I wonder how much guaranteed the order of the items in the output is. Let's believe the tests will catch it when it changes and then we can parse the lines into a hash table like we do with LVM info.

Reply:
I think the output won't change here, reiserfs is pretty old and I assume all development happens in reiser5. We are parsing NTFS, vfat and f2fs outputs in similar way.